crates/red_knot_python_semantic/resources/mdtest/unpacking.md

```diff
@@ -306,3 +306,110 @@ reveal_type(b)  # revealed: Unknown
 reveal_type(a)  # revealed: LiteralString
 reveal_type(b)  # revealed: LiteralString
 ```
+
+## Union
+
+### Same types
+
+Union of two tuples of equal length and each element is of the same type.
+
+```py
+def _(arg: tuple[int, int] | tuple[int, int]):
```

Comment:
I'm pretty sure this will immediately simplify to just `tuple[int, int]`, so it's not really testing unpacking of a union, just union simplification, but that's fine; it's still relevant to test the overall behavior.